app/components/quiz/Quiz.js

```diff
@@ -32,20 +32,56 @@ const Quiz = React.createClass({
   getInitialState: function() {
     var questionFlags = {}
     for (var i = 0; i < this.props.questions.length; i++) {
-      questionFlags[this.props.questions[i].id] = false
+      if (this.props.questions[i].type !== 'text') {
+        questionFlags[this.props.questions[i].id] = false
+      }
     }
-    return {questionAnswerFlags: questionFlags};
+    //console.log(questionFlags);
+    return {questionAnswerFlags: questionFlags, answerText: {}};
   },
 
-  onUpdate: function(questionId, onAnswered) {
-    var questionFlags = this.state.questionAnswerFlags;
-    questionFlags[questionId] = onAnswered;
-    this.setState({questionAnswerFlags: questionFlags});
+  onUpdate: function(questionId, onAnswered, text=null) {
+    if (text === null) {
+      var questionFlags = this.state.questionAnswerFlags;
+      questionFlags[questionId] = onAnswered;
+      this.setState({questionAnswerFlags: questionFlags});
+    } else {
+      //console.log(text);
+      var answerText = this.state.answerText === null ? {} : this.state.answerText;
+      answerText[questionId] = text;
+      this.setState({answerText: answerText});
+      //console.log(answerText);
+      //need to store in state
+    }
+  },
+
+  textQuestionClickNext: function() {
```

Comment:
Yeah sorry, I was trying to make it more general to incorporate different types of questions.
It basically updates the state of the answers. The reason I split it up based on the "text" parameter, is that I'm treating radio/checkbox answers as non-text, and storing them as flags, whereas the text and datetime (in the other PR) are just stored as text.